### PR TITLE
Adapt to new Market Api 1.6.1

### PR DESCRIPTION
**Pull request by tworec**
Conform to changes in https://github.com/golemfactory/ya-client/pull/52

### CODE REVIEW

File: agent/provider/src/market/mock_negotiator.rs [hunk 1]
```diff
@@ -87,15 +87,25 @@ impl Negotiator for LimitAgreementsNegotiator {
                 "Negotiator: Reject proposal [{:?}] due to expiration limits.",
                 demand.proposal_id
             );
-            Ok(ProposalResponse::RejectProposal)
+            Ok(ProposalResponse::RejectProposal {
+                reason: Some(format!(
+                    "proposal expired at: {} which is less than 5 min or more than 30 min from now",
```
Comment (nieznanysprawiciel) [inline agent/provider/src/market/mock_negotiator.rs:92]:
```suggestion
                    "Proposal expires at: {} which is less than 5 min or more than 30 min from now",
```

**Review by jiivan — APPROVED** [review]
lgtm

**Review by Wiezzel — APPROVED** [review]
Looks fine, but you gotta merge golemfactory/ya-client#52 ;)

File: agent/provider/src/market/mock_negotiator.rs [hunk 2]
```diff
@@ -87,15 +87,25 @@ impl Negotiator for LimitAgreementsNegotiator {
...
             Ok(ProposalResponse::AcceptProposal)
         } else {
             log::info!(
                 "Negotiator: Reject proposal [{:?}] due to limit.",
                 demand.proposal_id
             );
-            Ok(ProposalResponse::RejectProposal)
+            Ok(ProposalResponse::RejectProposal {
+                reason: Some(format!(
+                    "available agreements limit: {} reached",
```
Comment (nieznanysprawiciel) [inline agent/provider/src/market/mock_negotiator.rs:105]:
```suggestion
                    "Node busy. Reached Agreements limit: {}",
```

File: agent/provider/src/market/mock_negotiator.rs [hunk 3]
```diff
@@ -109,7 +119,12 @@ impl Negotiator for LimitAgreementsNegotiator {
                 "Negotiator: Reject agreement proposal [{}] due to limit.",
                 agreement.agreement_id
             );
-            Ok(AgreementResponse::RejectAgreement)
+            Ok(AgreementResponse::RejectAgreement {
+                reason: Some(format!(
+                    "available agreements limit: {} reached",
```
Comment (nieznanysprawiciel) [inline agent/provider/src/market/mock_negotiator.rs:124]:
```suggestion
                    "Node busy. Reached Agreements limit: {}",
```

File: agent/provider/src/market/negotiator.rs [hunk 4]
```diff
@@ -16,7 +16,10 @@ pub enum ProposalResponse {
         offer: DemandOfferBase,
     },
     AcceptProposal,
-    RejectProposal,
+    #[display(fmt = "RejectProposal( reason: {:?})", reason)]
```
Comment (nieznanysprawiciel) [inline agent/provider/src/market/negotiator.rs:19]: It will be awfull message with Some()

**Review by nieznanysprawiciel — APPROVED** [review]

File: agent/provider/src/market/negotiator.rs (hunk 4 above)
Comment (tworec) [reply agent/provider/src/market/negotiator.rs:19]: Let's cover it in other PR, as this is blocking others.

Comment (tworec) [reply agent/provider/src/market/negotiator.rs:19]: reported as #776

**Pull request merged**